shared/TestStandComm/SerialSession.cxx

```diff
@@ -1,65 +1,130 @@
 #include "SerialSession.h"
 
+/** Maximum time to wait to receive an ACK after sending a message (milliseconds) */
 #define ACK_TIMEOUT_MS 500
 
+/**
+ * @brief Constructs a new SerialSession
+ * 
+ * @param transport The underlying SerialTransport to use for sending and receiving messages
+ * @param received_msg A reference to a Message struct where received messages will be placed
+ */
 SerialSession::SerialSession(SerialTransport& transport, Message& received_msg) : received_msg(received_msg), transport(transport)
 {
     // Nothing else to do
 }
 
-void SerialSession::ack()
+/**
+ * @brief Transmits an ACK message
+ */
+bool SerialSession::ack()
 {
     Message ack = {
         .id = MSG_ID_ACK,
         .length = 0,
         .data = nullptr
     };
-    this->transport.send_message(ack);
+    return this->transport.send_message(ack);
 }
 
-bool SerialSession::check_for_message()
+/**
+ * @brief Checks if the received message was expected and sends an ACK if it was
```

Comment:
Is `SERIAL_OK` an `ACK`? Since you say that we return an `ACK` on success, but actually return `SERIAL_OK`

Reply:
We **send** an ACK (like send it over serial). We **return** `SERIAL_OK`